src/point_one/fusion_engine/messages/configuration.h

```diff
@@ -2221,6 +2242,62 @@ inline p1_ostream& operator<<(p1_ostream& stream, MessageRate val) {
   return stream;
 }
 
+/**
+ * @brief TCP client/server interface configuration parameters.
+ * @ingroup io_interfaces
+ */
+struct P1_ALIGNAS(4) TCPConfig {
+  uint8_t enabled = 1;
+  TransportDirection direction = TransportDirection::SERVER;
+  uint16_t port = 0;
+  char remote_address[64] = {0};
```

Comment:
its odd to call this remote address if this represents a server and a client.  My personal opinion is that TCPServer and TCPClient should be treated as two different top level entities. There is no such thing as a remote address on a TCP server and if its intended to be the bind_address, it should be named as such.